test/ruby/test_zjit.rb

```diff
@@ -1664,6 +1664,28 @@ def test() = @foo = 1
     }
   end
 
+  def test_getclassvariable
+    assert_compiles '42', %q{
+      class Foo
+        def self.test = @@x
+      end
+
+      Foo.class_variable_set(:@@x, 42)
+      Foo.test()
+    }
+  end
+
+  def test_setclassvariable
```

Comment:
it would be good to test also something like raising an exception from a classvariable set (the class could be frozen) to exercise the `non_leaf_call` bit. something like:

```
irb(main):001* class C
irb(main):002*   def test
irb(main):003*     @@x = 5
irb(main):004*   end
irb(main):005> end
=> :test
irb(main):006> C.freeze
=> C
irb(main):009> C.new.test
(irb):3:in 'C#test': can't modify frozen #<Class:C>: C (FrozenError)
        from (irb):9:in '<main>'
        from <internal:kernel>:168:in 'Kernel#loop'
        from /Users/emacs/.gem/ruby/ruby-3.4.3/gems/irb-1.15.2/exe/irb:9:in '<top (required)>'
        from /Users/emacs/.gem/ruby/ruby-3.4.3/bin/irb:25:in 'Kernel#load'
        from /Users/emacs/.gem/ruby/ruby-3.4.3/bin/irb:25:in '<main>'
irb(main):010>
```